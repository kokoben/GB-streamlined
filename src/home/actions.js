const HOME_VIDEO_SET = 'HOME_VIDEO_SET';
export const HOME_VIDEO_SET_SUCCESS = 'HOME_VIDEO_SET_SUCCEEDED';
export const HOME_VIDEO_SET_FAIL = 'HOME_VIDEO_SET_FAILED';

export const setHomeVideo = () => {
  console.log('inside setHomeVideo');
  return {
    type: HOME_VIDEO_SET,
  }
}
