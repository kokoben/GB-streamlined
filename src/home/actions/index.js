import * as hActions from './types';

export const setHomeVideo = (id) => {
  console.log('inside setHomeVideo');
  return {
    type: hActions.HOME_VIDEO_SET,
    id,
  };
};

export const setHomeVideos = (page, pageSize) => {
  console.log('inside setHomeVideos');
  return {
    type: hActions.HOME_VIDEOS_SET,
    page,
    pageSize,
  };
};

export const setLatestHomeVideo = () => ({
  type: hActions.LATEST_HOME_VIDEO_SET,
});

export const fetchHomeSearchVideos = keywords => ({
  type: hActions.HOME_SEARCH_FETCH,
  keywords,
});
