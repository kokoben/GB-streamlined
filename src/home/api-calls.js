import { apiKey } from '../api-keys';

// get latest video
export const requestHomeVideo = (id) => {
  return "https://www.giantbomb.com/api/video/" +
    id + "/?api_key=" + apiKey +
    "&format=jsonp&json_callback=callback&field_list=name," +
    "id,deck,embed_player,publish_date,image,video_type,hd_url,user";
}

export const requestHomeVideos = page => {
  return "https://www.giantbomb.com/api/videos/" +
    "?api_key=" + apiKey +
    "&format=jsonp&json_callback=callback&limit=24&offset=" +
    (page-1) * 24 +
    "&field_list=hd_url,name,id,deck,embed_player,publish_date,image,video_type," +
    "user&filter=video_type:3|8|23&sort=publish_date:desc";
}
