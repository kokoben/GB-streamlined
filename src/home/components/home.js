import React from 'react';
import CurrentHomeVideo from '../containers/current-home-video';
import HomeVideos from '../containers/home-videos';

export default () => (
  <div>
    <CurrentHomeVideo />
    <HomeVideos />
  </div>
)
