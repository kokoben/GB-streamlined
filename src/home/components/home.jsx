import React from 'react';
import CurrentHomeVideo from '../containers/current-home-video';
import HomeVideos from '../containers/home-videos';
import HomeFeedHeader from '../components/home-feedheader';

export default () => (
  <div>
    <CurrentHomeVideo />
    <HomeFeedHeader />
    <HomeVideos />
  </div>
);
