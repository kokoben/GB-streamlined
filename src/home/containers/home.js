import React from 'react';
import CurrentHomeVideo from './current-home-video';
import HomeVideos from './home-videos';

export default () => (
  <div>
    <CurrentHomeVideo />
    <HomeVideos />
  </div>
)
