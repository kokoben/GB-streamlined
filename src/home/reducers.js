import { 
  HOME_VIDEO_SET_SUCCESS, 
  HOME_VIDEO_SET_FAIL,
  HOME_VIDEOS_SET_SUCCESS,
  HOME_VIDEOS_SET_FAIL,
  LATEST_HOME_VIDEO_SET_SUCCESS,
  LATEST_HOME_VIDEO_SET_FAIL
} from './actions';

export const homeVideo = (state=null, action) => {
  switch(action.type) {
    case HOME_VIDEO_SET_SUCCESS:
      return action.response;
    case HOME_VIDEO_SET_FAIL:
      console.log('home_video_set_fail!');
      console.log(action.message);
      break;
    default:
      return state;
  }
}

export const homeVideos = (state=null, action) => {
  switch(action.type) {
    case HOME_VIDEOS_SET_SUCCESS:
      console.log('home_video_set_success!');
      const data = action.response;
      return data;
    case HOME_VIDEOS_SET_FAIL:
      console.log('home_video_set_fail!');
      console.log(action.message);
      break;
    default:
      return state;
  }
}
