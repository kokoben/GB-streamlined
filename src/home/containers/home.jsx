import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import CurrentHomeVideoConnected from '../containers/current-home-video';
import HomeVideosConnected from '../containers/home-videos';
import HomeFeedHeaderConnected from '../components/home-feedheader';
import { setSearchPage, setSearchMarker } from '../../actions/index';

export class Home extends Component {
  constructor(props) {
    super(props);

    this.onClickPageScroll = this.onClickPageScroll.bind(this);
  }

  componentDidMount() {
    this.props.setSearchPage(null);
    this.props.setSearchMarker(null);
  }

  onClickPageScroll() {
    this.feedHeaderNode.scrollIntoView();
  }

  render() {
    return (
      <div>
        <CurrentHomeVideoConnected />
        <div ref={node => this.feedHeaderNode = node}>
          <HomeFeedHeaderConnected />
        </div>
        <HomeVideosConnected onClickPageScroll={this.onClickPageScroll} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setSearchPage,
    setSearchMarker,
  }, dispatch)
);

Home.propTypes = {
  setSearchPage: PropTypes.func.isRequired,
  setSearchMarker: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(Home);
