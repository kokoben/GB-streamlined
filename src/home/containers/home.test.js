import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Home } from './home';

it('renders correctly', () => {
  const mockSetSearchPage = jest.fn();
  const wrapper = shallow(<Home
    setSearchPage={mockSetSearchPage}
  />);

  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
