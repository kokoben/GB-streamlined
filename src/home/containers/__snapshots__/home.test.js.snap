// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Connect(CurrentHomeVideo) />
  <Connect(Component) />
  <Connect(HomeVideos) />
</div>
`;
